### ISSUE TITLE
provide a way to ignore master_authorized_networks when using the beta-private-cluster with GCP's cloud build

### ISSUE THREAD

**Issue by sreenivas-ps** [terraform-google-modules/terraform-google-kubernetes-engine#1184]
### TL;DR

I'm using the terraform community gke module `beta-private-cluster` v17.3.0 with GCPs Cloud Build (CB) as our CI. Since this is a private cluster everytime CB CI runs it uses a new external IP and it cannot connect to the cluster. To get around it I get the CB's IP, add it as first step to the mater authorised networks, then run the terraform step, and remove the IP added as the last step of the build. Since CB's IP is dynamic, we cannot add the IP to the `master_authorized_networks` as a config; during the CI terraform apply step, it is removing the IP added to the cluster (added as the first step) as terraform thinks it not the source config, which updates the cluster everytime we run terraform apply, and later the deployment may fail since it removed the CB's IP. I wonder if we can add an additional variable (possibly as list) to configure to concat this variable to the line here: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/v17.3.0/modules/beta-private-cluster/cluster.tf#L220

As a workaround, I have to download this version of the module into our repo and append the `master_authorized_networks` to the link ^.

### Terraform Resources

```markdown
google_container_cluster

https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/v17.3.0/modules/beta-private-cluster/cluster.tf#L220
```

### Additional information

terraform version: 0.14.8
hashicorp/google-beta v3.90.1.

**Comment by bharathkkb** [comment]
Thanks for the FR @sreenivas-ps 
Have you considered Cloud Build [private pools](https://cloud.google.com/build/docs/private-pools/private-pools-overview#network_architecture) which will remove the need to keep updating the authorized networks. You can see a tutorial [here](https://cloud.google.com/architecture/accessing-private-gke-clusters-with-cloud-build-private-pools) and an example [terraform implementation](https://github.com/GoogleCloudPlatform/terraform-google-secure-cicd/tree/1296c3238f43f11eb5600086f5418d5ae91933ed/examples/gke_cloudbuild_private_pool) here.

**Comment by sreenivas-ps** [comment]
@bharathkkb : I've already implemented private pools for projects in production but this setup of private pools with VPN is very expensive to do the same for the dev, stg envs. are you considering to implement this FR ?

**Comment by bharathkkb** [comment]
@sreenivas-ps sorry for the delay. Can you leverage https://cloud.google.com/anthos/multicluster-management/gateway/setup instead? Here is an example with cloud build https://cloud.google.com/anthos/multicluster-management/gateway/tutorials/cloud-build-integration
Regarding FR, I believe we cannot add this optionally since lifecycle rules are static and in most cases you do want to track this field so I am inclined to not add this.
